### PR TITLE
DMN test cases for hit policies

### PR DESCRIPTION
**Pull request by AggarwSh**

### CODE REVIEW

**Comment by DanielThanner** [comment]
@AggarwSh Can you please change the filenames of all xml files to *-test-01.xml (same filename format as for existing tests)?

I think there are some issues with the tests (0117, 0118, 0119):
- As mentioned in the DMN 1.1 spec on page 83: "Decision tables with compound outputs support only the following hit policies: Unique, Any, Priority, First, Output
order, Rule order and Collect without operator, because the collect operator is undefined over multiple outputs." Therefore the output of the decisoin table should be null, since the configuration of the decision table itself is invalid. You can't use an aggregation function in conjunction with a decision table that has multiple output clauses.

I think the other testcases (0108-0116) are compliant with the DMN 1.1 specification. They run successfully with our engine. But the xml filename should be changed (see above).

**Comment by AggarwSh** [comment]
@DanielThanner I have done the changes as suggested by you.

Please review and let me know in case of any issue.

File: TestCases/compliance-level-2/0114-min-collect-hitpolicy/0114-min-collect-hitpolicy-test-01.xml [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+	</labels>
+	<testCase id="001">
+		<inputNode name="NumOfYears">
+			<value>5</value>
+		</inputNode>
+		<resultNode name="CarInsurance" type="decision">
+			<expected>
+			<list>
+				<item>
+					<value>64.32</value>
```
Comment (etirelli) [inline TestCases/compliance-level-2/0114-min-collect-hitpolicy/0114-min-collect-hitpolicy-test-01.xml]: This should be a single value, not a list, as the C< hit policy returns a single value.

File: TestCases/compliance-level-2/0114-min-collect-hitpolicy/0114-min-collect-hitpolicy-test-01.xml [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+	</testCase>
+	<testCase id="002">
+		<inputNode name="NumOfYears">
+			<value>3</value>
+		</inputNode>
+		<resultNode name="CarInsurance" type="decision">
+			<expected>
+			<list>
+				<item>
+					<value>98.83</value>
```
Comment (etirelli) [inline TestCases/compliance-level-2/0114-min-collect-hitpolicy/0114-min-collect-hitpolicy-test-01.xml]: Same here.

File: TestCases/compliance-level-2/0114-min-collect-hitpolicy/0114-min-collect-hitpolicy-test-01.xml [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+	</testCase>
+	<testCase id="003">
+		<inputNode name="NumOfYears">
+			<value>4</value>
+		</inputNode>
+		<resultNode name="CarInsurance" type="decision">
+			<expected>
+			<list>
+				<item>
+					<value>98.83</value>
```
Comment (etirelli) [inline TestCases/compliance-level-2/0114-min-collect-hitpolicy/0114-min-collect-hitpolicy-test-01.xml]: Same here.

File: TestCases/compliance-level-2/0115-sum-collect-hitpolicy/0115-sum-collect-hitpolicy-test-01.xml [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+	</labels>
+	<testCase id="001">
+		<inputNode name="NumOfYears">
+			<value>6</value>
+		</inputNode>
+		<resultNode name="Salary" type="decision">
+			<expected>
+			<list>
+				<item>
+					<value>1100</value>
```
Comment (etirelli) [inline TestCases/compliance-level-2/0115-sum-collect-hitpolicy/0115-sum-collect-hitpolicy-test-01.xml]: This should be a single value, not a list.

File: TestCases/compliance-level-2/0115-sum-collect-hitpolicy/0115-sum-collect-hitpolicy-test-01.xml [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+	</testCase>
+	<testCase id="002">
+		<inputNode name="NumOfYears">
+			<value>3</value>
+		</inputNode>
+		<resultNode name="Salary" type="decision">
+			<expected>
+			<list>
+				<item>
+					<value>300</value>
```
Comment (etirelli) [inline TestCases/compliance-level-2/0115-sum-collect-hitpolicy/0115-sum-collect-hitpolicy-test-01.xml]: This should be a single value, not a list.

File: TestCases/compliance-level-2/0115-sum-collect-hitpolicy/0115-sum-collect-hitpolicy-test-01.xml [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+	</testCase>
+	<testCase id="003">
+		<inputNode name="NumOfYears">
+			<value>2</value>
+		</inputNode>
+		<resultNode name="Salary" type="decision">
+			<expected>
+			<list>
+				<item>
+					<value>100</value>
```
Comment (etirelli) [inline TestCases/compliance-level-2/0115-sum-collect-hitpolicy/0115-sum-collect-hitpolicy-test-01.xml]: This should be a single value, not a list.

File: TestCases/compliance-level-2/0116-count-collect-hitpolicy/0116-count-collect-hitpolicy-test-01.xml [hunk 7]
```diff
@@ -0,0 +1,54 @@
...
+	</labels>
+	<testCase id="001">
+		<inputNode name="NumOfYears">
+			<value>6</value>
+		</inputNode>
+		<resultNode name="Salary" type="decision">
+			<expected>
+			<list>
+				<item>
+					<value>4</value>
```
Comment (etirelli) [inline TestCases/compliance-level-2/0116-count-collect-hitpolicy/0116-count-collect-hitpolicy-test-01.xml]: This should be a single value, not a list.

File: TestCases/compliance-level-2/0116-count-collect-hitpolicy/0116-count-collect-hitpolicy-test-01.xml [hunk 8]
```diff
@@ -0,0 +1,54 @@
...
+	</testCase>
+	<testCase id="002">
+		<inputNode name="NumOfYears">
+			<value>2</value>
+		</inputNode>
+		<resultNode name="Salary" type="decision">
+			<expected>
+			<list>
+				<item>
+					<value>1</value>
```
Comment (etirelli) [inline TestCases/compliance-level-2/0116-count-collect-hitpolicy/0116-count-collect-hitpolicy-test-01.xml]: This should be a single value, not a list.

File: TestCases/compliance-level-2/0116-count-collect-hitpolicy/0116-count-collect-hitpolicy-test-01.xml [hunk 9]
```diff
@@ -0,0 +1,54 @@
...
+	</testCase>
+	<testCase id="003">
+		<inputNode name="NumOfYears">
+			<value>3</value>
+		</inputNode>
+		<resultNode name="Salary" type="decision">
+			<expected>
+			<list>
+				<item>
+					<value>2</value>
```
Comment (etirelli) [inline TestCases/compliance-level-2/0116-count-collect-hitpolicy/0116-count-collect-hitpolicy-test-01.xml]: This should be a single value, not a list.

**Comment by AggarwSh** [comment]
@etirelli I have incorporated the changes suggested by you. Please review

**Comment by DanielThanner** [comment]
@AggarwSh Our DMN runtime executes all testcases without any errors. Well done.

**Pull request merged**